### PR TITLE
Decimal v6

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

**Comment by losipiuk** [comment]
This supersedes https://github.com/facebook/presto/pull/4626

**Comment by losipiuk** [comment]
This is new version with applied comments from v5.
@martint, there are still a few open questions to you in v5 - so please take a look.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java [hunk 1]
```diff
@@ -289,9 +295,9 @@ private static TypeSignature getTypeSignature(TypeInfo typeInfo)
         throw new PrestoException(NOT_SUPPORTED, format("Unsupported Hive type: %s", typeInfo));
     }
 
-    private static Type getPrimitiveType(PrimitiveObjectInspector.PrimitiveCategory primitiveCategory)
+public static Type getPrimitiveType(PrimitiveTypeInfo typeInfo)
```
Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java]: formatting

**Comment by martint** [comment]
Typo in commit message for "Use isTypeOnlyCoercion for Type arguemnts when possible".

Also, add a comment in the commit message explaining why.

File: presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignatureParameter.java [hunk 2]
```diff
@@ -125,10 +125,14 @@ public boolean isCalculated()
         switch (kind) {
             case TYPE:
                 return getTypeSignature().isCalculated();
+            case NAMED_TYPE:
```
Comment (martint) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignatureParameter.java]: Add a comment to the commit message explaining what was broken

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java (hunk 1 above)
Comment (arhimondr) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java]: done

File: presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignatureParameter.java (hunk 2 above)
Comment (arhimondr) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignatureParameter.java]: done

**Comment by arhimondr** [comment]
> Typo in commit message for "Use isTyp ...

Done

**Comment by arhimondr** [comment]
Superseded by: #4810

**Comment by losipiuk** [comment]
superseded by https://github.com/facebook/presto/pull/4810

**Pull request closed without merging**